antarest/study/service.py

```diff
@@ -213,17 +228,37 @@ def _generate_timeseries(self) -> None:
                 )
             )
 
-    def run_task(self, notifier: TaskUpdateNotifier) -> TaskResult:
+    def run_task(self, notifier: TaskUpdateNotifier, listener: t.Optional[ICommandListener]) -> TaskResult:
         msg = f"Generating thermal timeseries for study '{self._study_id}'"
         notifier(msg)
-        self._generate_timeseries()
+        self._generate_timeseries(listener)
         msg = f"Successfully generated thermal timeseries for study '{self._study_id}'"
         notifier(msg)
         return TaskResult(success=True, message=msg)
 
     # Make `ThermalClusterTimeSeriesGeneratorTask` object callable
     __call__ = run_task
 
+    class TsGenerationListener(ICommandListener):
```

Comment:
Finally, we don't need it here:
we should instead make an evolution of `TaskJobLogRecorder` so that it does this for all types of tasks.

Then the command listener implementation we will need to implement is just a forwarding of the progress to the task listener.